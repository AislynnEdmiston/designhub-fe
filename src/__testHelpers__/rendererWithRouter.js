import React from 'react'
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

import rootReducer from '../store/reducers';
jest.mock('../auth-wrapper');

const store = createStore(
  rootReducer,
  applyMiddleware(thunk)
);
const rendererWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Provider store={store}>
        <Router history={history}>
          {ui}
        </Router>
      </Provider>
    ),
    history
  }
}

export default rendererWithRouter;
