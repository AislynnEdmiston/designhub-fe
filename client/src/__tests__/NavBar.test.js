import React from 'react'
import NavBar from '../components/NavBar'

import rendererWithRouter from '../__testHelpers__/rendererWithRouter'

describe('<NavBar />', () => {

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const activeUser = {
        id: 1
      }
      const tree = rendererWithRouter(
        <NavBar activeUser={activeUser} history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
})
