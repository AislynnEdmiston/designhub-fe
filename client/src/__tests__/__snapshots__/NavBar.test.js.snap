// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<nav>
  <div
    className="icon-links"
  >
    <div
      className="navlinks"
    >
      <a
        aria-current={null}
        href="/profile/1/undefined"
        onClick={[Function]}
      >
        <div
          className="ProfileIcon"
        >
          <svg
            viewBox="0 0 21 21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.2926 17.7751V16.1084C17.2926 15.2244 16.9414 14.3765 16.3162 13.7514C15.6911 13.1263 14.8433 12.7751 13.9592 12.7751H7.29256C6.40851 12.7751 5.56066 13.1263 4.93554 13.7514C4.31042 14.3765 3.95923 15.2244 3.95923 16.1084V17.7751"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M10.6261 9.44178C12.467 9.44178 13.9594 7.9494 13.9594 6.10845C13.9594 4.2675 12.467 2.77512 10.6261 2.77512C8.78511 2.77512 7.29272 4.2675 7.29272 6.10845C7.29272 7.9494 8.78511 9.44178 10.6261 9.44178Z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </svg>
        </div>
      </a>
    </div>
    <div
      className="navlinks"
    >
      <a
        aria-current={null}
        href="/create"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="21"
          viewBox="0 0 21 21"
          width="21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.4593 2.58347H4.79264C3.87217 2.58347 3.12598 3.32966 3.12598 4.25013V15.9168C3.12598 16.8373 3.87217 17.5835 4.79264 17.5835H16.4593C17.3798 17.5835 18.126 16.8373 18.126 15.9168V4.25013C18.126 3.32966 17.3798 2.58347 16.4593 2.58347Z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
          <path
            d="M10.626 6.75013V13.4168"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
          <path
            d="M7.29272 10.0835H13.9594"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
        </svg>
      </a>
    </div>
    <div
      className="navlinks"
    >
      <a
        aria-current={null}
        href="/explore"
        onClick={[Function]}
      >
        <svg
          className="feather feather-compass"
          fill="none"
          height="20"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <polygon
            points="16.24 7.76 14.12 14.12 7.76 16.24 9.88 9.88 16.24 7.76"
          />
        </svg>
      </a>
    </div>
    <div
      className="navlinks"
    >
      <a
        aria-current={null}
        href="/notifications/1/undefined"
        onClick={[Function]}
      >
        <svg
          height="21"
          viewBox="0 0 20 21"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.3334 10.7001H13.3334L11.6667 13.2001H8.33342L6.66675 10.7001H1.66675"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
          <path
            d="M4.54175 4.95848L1.66675 10.7001V15.7001C1.66675 16.1422 1.84234 16.5661 2.1549 16.8787C2.46746 17.1912 2.89139 17.3668 3.33341 17.3668H16.6667C17.1088 17.3668 17.5327 17.1912 17.8453 16.8787C18.1578 16.5661 18.3334 16.1422 18.3334 15.7001V10.7001L15.4584 4.95848C15.3204 4.6808 15.1077 4.44712 14.8442 4.28371C14.5807 4.12031 14.2768 4.03365 13.9667 4.03348H6.03341C5.72334 4.03365 5.41947 4.12031 5.15595 4.28371C4.89244 4.44712 4.67973 4.6808 4.54175 4.95848V4.95848Z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
        </svg>
        
      </a>
    </div>
    <div
      className="navlinks"
    >
      <a
        aria-current={null}
        href="/settings"
        onClick={[Function]}
      >
        <svg
          className="feather feather-settings"
          fill="none"
          height="20"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="text-links"
  >
    <a
      aria-current={null}
      className="links"
      href="/profile/1/undefined"
      onClick={[Function]}
    >
      Profile
    </a>
    <a
      aria-current={null}
      className="links"
      href="/create"
      onClick={[Function]}
    >
      New Project
    </a>
    <a
      aria-current={null}
      className="links"
      href="/explore"
      onClick={[Function]}
    >
      Explore
    </a>
    <a
      aria-current={null}
      className="links"
      href="/notifications/1/undefined"
      onClick={[Function]}
    >
      <span
        onClick={[Function]}
      >
        Notifications
      </span>
    </a>
    <a
      aria-current={null}
      className="links"
      href="/settings"
      onClick={[Function]}
    >
      Settings
    </a>
  </div>
</nav>
`;
